packages/mwp-tracking-plugin/src/_activityTrackers.js

```diff
@@ -48,29 +49,19 @@ export const getTrackApiResponses: TrackGetter = trackOpts => request => (
  *    - url: proxy endpoint path (request.url.path)
  *    - referrer: current URL (request.referrer)
  */
-export const getTrackActivity: TrackGetter = () => request => (
+export const getTrackActivity: TrackGetter = () => (request: HapiRequest) => (
 	fields: ActivityFields
 ) => {
-	const { url, method, payload, query, info: { referrer } } = request;
-	const requestReferrer = referrer || '';
-	const reqData = method === 'post' ? payload : query;
-
-	// the request may specify a referrer that should be used instead of the `request.referrer`
-	const referrerOverride =
-		reqData.metadata && (rison.decode_object(reqData.metadata) || {}).referrer;
-
-	const urlFields = referrerOverride
-		? {
-				url: requestReferrer, // original request referrer is passed through as URL
-				referrer: referrerOverride,
-			}
+	// route may specify a custom 'getFields', which usually means that it is a
+	// proxy endpoint that should be tracked differently
+	const { getFields } =
+		request.route.settings.plugins[ACTIVITY_PLUGIN_NAME] || {};
+	const trackFields = getFields
+		? getFields(request, fields)
```

Comment:
Just verifying that in this special case, we don't care about explicitly tracking the url and the referrer any longer?

Reply:
correct - `getFields` is expected to fully handle those values. I'll add some docs before merging